.github/workflows/ci.yml

```diff
@@ -444,14 +445,11 @@ jobs:
     if: ${{ inputs.enable-coverage }}
     runs-on: ubuntu-latest
     steps:
-      - name: Retrieve code coverage statuses
-        uses: actions/download-artifact@v3
-        with:
-          path: status
-      # Check that there are no incomplete tests.
+      # Check that there are no failed tests.
       - name: Check tests completed successfully.
         run: |
-          if test -z "$(grep -rl 'incomplete' ./status)"; then
+          # turn this into an if
+          if test -z "$(echo ${{ toJson(needs) }} | jq -r '.[] | .outputs | .result' | grep -v 'success')"; then
```

Comment:
```suggestion
          JSON='${{ toJson(needs) }}'
          if test -z "$(echo $JSON | jq -r '.[] | .outputs | .result' | grep -v 'success')"; then
```

Reply:
In the debug output in https://github.com/pulumi/pulumi/actions/runs/9704560770/job/26785864891#step:2:126 we have two single quotes around the strings in the test: `jq -r ''.[] | .outputs | .result''`, I am not sure what's up with that.